src/front-door/azext_front_door/_params.py

```diff
@@ -122,6 +122,10 @@ def load_arguments(self, _):
             c.argument('http_port', type=int, help='HTTP TCP port number.')
             c.argument('https_port', type=int, help='HTTPS TCP port number.')
             c.argument('weight', type=int, help='Weight of this endpoint for load balancing purposes.')
+            c.argument('private_link_alias', help='The Alias of the Private Link resource. Populating this optional field indicates that this backend is \'Private\'.')
+            c.argument('private_link_resource_id', help='The Resource Id of the Private Link. Populating this optional field indicates that this backend is \'Private\'.')
+            c.argument('private_link_location', help='The location of the Private Link resource. Required only if \'privateLinkResourceId\' is populated.')
```

Comment:
what's the meaning of location? Can user choose one location?

Reply:
The location designates the location of the private link to which a frontdoor can connect. While the location can be any location that is supported by Frontdoor private link feature, the user will almost always choose the location of or closest to the private link provided to minimize latency of the request consuming said service behind the private link

Reply:
I thought user need to provide private link resource id already, why do they need support location one more time?

Reply:
Once a user has submitted a frontdoor with a private backend (by providing alias or resource id/location), Frontdoor create a private endpoint in a Frontdoor-owned virtual network. Requests coming to this private backend will be forwarded through said private endpoint, having it closer to the private link lowers the latency for the request.

Also, not all private links have locations, some can be global. It's up to the user to pick the location that makes sense to their use case, but it doesn't have to be a particular value